packages/service-worker/worker/src/driver.ts

```
@@ -739,6 +746,18 @@ export class Driver implements Debuggable, UpdateSource {
           appVersion = this.lookupVersionByHash(this.latestHash, 'assignVersion');
         }
 
+        if (isWorkerScriptRequest) {
+          if (!this.clientVersionMap.has(event.resultingClientId)) {
+            // New worker hasn't been seen before; set this client to requesting client version
+            this.clientVersionMap.set(event.resultingClientId, hash);
+            await this.sync();
+          } else if (this.clientVersionMap.get(event.resultingClientId)! !== hash) {
+            throw new Error(
+              `Version mismatch between worker client ${event.resultingClientId} and requesting client ${clientId}`,
+            );
```

I'm not sure if we should throw an error, or fallback to network via `return null;`